.github/workflows/makefile.yml

```diff
@@ -0,0 +1,280 @@
+
+  nitunit-full:
+    runs-on: ubuntu-24.04
+    needs: [nitunit-some]
+    container: "ubuntu:24.04"
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+      mongo:
+        image: mongo
+      neo4j:
+        image: 'neo4j:2.3'
+        env:
+          NEO4J_AUTH: none
+    steps:
+    - name: Install dependencies
+      run: apt-get update && apt-get install -y $BASIC_PACKAGES $MORE_PACKAGES
+    - uses: actions/checkout@v4
+    - run: misc/jenkins/ensuregit.sh
+    - uses: actions/download-artifact@v4
+    - run: chmod +x bin/*
+    - run: make nitunit-full
+    - uses: actions/upload-artifact@v4
+      if: success() || failure()
+      with:
+        path: '*.xml'
+        name: nitunit-full
+    - uses: EnricoMi/publish-unit-test-result-action@v2
+      if: success() || failure()
+      with:
+        files: "*.xml"
+        comment_mode: off
+        check_name: 'nitunit-full Results'
+
+  test-contrib:
+    runs-on: ubuntu-24.04
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES $MORE_PACKAGES
+    - uses: actions/download-artifact@v4
+    - run: chmod +x bin/*
+    - run: 'PATH=$PWD/bin:$PATH make test-contrib'
+    - uses: actions/upload-artifact@v4
+      if: success() || failure()
+      with:
+        path: '*.xml'
+        name: test-contrib
+    - uses: EnricoMi/publish-unit-test-result-action@v2
+      if: success() || failure()
+      with:
+        files: "*.xml"
+        comment_mode: off
+        check_name: 'test-contrib Results'
+
+  bootstrap-full:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES
+    - run: make bootstrap-full
+
+  bench-fast:
+    runs-on: ubuntu-24.04
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES gnuplot
+    - uses: actions/download-artifact@v4
+    - run: chmod +x bin/*
+    - run: 'PATH=$PWD/bin:$PATH make bench-fast'
+    - uses: actions/upload-artifact@v4
+      if: success() || failure()
+      with:
+        path: |-
+          benchmarks/*.dat
+          benchmarks/*.gnu
+          benchmarks/*.png
+          benchmarks/*.html
+        name: bench-fast
+
+  valgrind:
+    runs-on: ubuntu-24.04
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES valgrind
+    - uses: actions/download-artifact@v4
+    - run: chmod +x bin/*
+    - run: make valgrind
+    - uses: actions/upload-artifact@v4
+      if: success() || failure()
+      with:
+        path: 'valgrind.out'
+        name: valgrind
+
+  info:
```

Comment:
Was this only for debug? Still needed?

Reply:
Is is more a health check of github action, for whatever they change in their infrastructure and setup. It will be easier to inspect what happened in case of breakage.